src/ServiceControl/Recoverability/Archiving/ArchiveOperationLogic.cs

```diff
@@ -0,0 +1,128 @@
+﻿namespace ServiceControl.Recoverability
+{
+    using ServiceControl.Infrastructure.DomainEvents;
+    using System;
+
+    public class ArchiveOperationLogic
```

Comment:
The problem is we store the `ArchiveOperation` as the header operation - which has the same name as this should have been. I'm open for other suggestions though?